tests/python_package_test/test_arrow.py

```diff
@@ -67,6 +67,10 @@ def dummy_dataset_params() -> Dict[str, Any]:
     }
 
 
+def assert_arrays_equal(lhs: np.ndarray, rhs: np.ndarray):
+    assert lhs.dtype == rhs.dtype and np.array_equal(lhs, rhs)
```

Comment:
Instead of this approach for testing if two `numpy` arrays are identical, I'd prefer that we use `np.testing.assert_array_equal(..., strict=True)`. That functionality is also so generically useful in `lightgbm`'s tests that I think it belongs in `tests/python_package_test/utils.py`.

I've introduced such a function in https://github.com/microsoft/LightGBM/pull/6108/files#diff-da8bd23ee10ae6337467b4d88500a4a695b154d23a17742a4cef514cb1d6cf85R206.

@jmoralez I'm waiting to merge that PR until you have a chance to review #6108 again, specifically https://github.com/microsoft/LightGBM/pull/6108#discussion_r1374048081. Will you have time to do that this week?

I know it's kind of involved, so if not I'll pull that new testing function out of that PR into aa separate one that'd hopefully require less reviewing effort, so we could re-use it here and in other tests.

Reply:
@jameslamb any chance #6108 is merged soon so I can take advantage of the new function? Or should I just copy the function into this branch so that we can proceed?

Reply:
Don't just copy that function into this branch... I want it to be reviewed by someone other than myself.

We can just merge this today and I'll clean up and consolidate the testing functions later.

Reply:
All right! I was under the impression that #6108 would have already been reviewed by @guolinke :smile:

Reply:
It has but @jmoralez and I have been collaborating on it (as can be seen in the comment thread I linked to from https://github.com/microsoft/LightGBM/pull/6163#discussion_r1382747360), so I don't want to merge it until he has the time to review.

Pull request reviews here aren't just quality gates like a building or restaurant inspection... they're a chance for maintainers to agree/disagree about the state of this thing we're all maintaining together. "at least one approval" is the minimum standard we enforce but that doesn't mean that every PR should be merged as soon as one person approves.

Reply:
Hey. Sorry for the delay, I moved last week and missed the update in #6108. I'm reviewing it now.

Reply:
No rush at all, don't let me add stress to your life! Seriously, it can happen any time.